README.md

```diff
@@ -1,5 +1,54 @@
+|--------------------------|----------|-------------------------------------------------------|--------------------------------------------------------------------------------------------------|-----------------------------|
+| consignment_id           | false    | N/A                                                   | TDR UUID for the consignment the object to scan belongs to                                       |                             |
+| file_id                  | false    | N/A                                                   | Name of the object to scan                                                                       |                             |
+| user_id                  | true     | `None`                                                | TDR UUID of the user who uploaded the object to scan                                             |                             |
+| original_path            | true     | `None`                                                | Original path to the object to scan. Used to create local version of object for scanning         |                             |
+| scan_type                | true     | N/A                                                   | **Deprecated**. Use combination of optional parameters to set configuration. Type of scan to run | `metadata` / `consignment`  |
+| s3_source_bucket         | true     | `tdr-upload-files-cloudfront-dirty-{tdr environment}` | S3 bucket containing the object to scan                                                          | `{some AWS S3 bucket name}` |
+| s3_source_bucket_key     | true     | `{user_id}/{consignment_id}/{file_id}`                | S3 bucket key of the object to scan                                                              |                             |
+| s3_source_bucket_key     | true     | `{user_id}/{consignment_id}/{file_id}`                | S3 bucket key of the object to scan                                                              |                             |
+| s3_upload_bucket         | true     | `tdr-upload-files-{tdr environment}`                  | S3 bucket to copy clean objects to                                                               |                             |
+| s3_upload_bucket_key     | true     | `{consignment_id}/{file_id}`                          | S3 bucket key of clean object                                                                    |                             |
+| s3_quarantine_bucket     | true     | `tdr-upload-files-quarantine-{tdr environment}`       | S3 bucket to copy infected objects to                                                            |                             |
+| s3_quarantine_bucket_key | true     | `{consignment_id}/{file_id}`                          | S3 bucket key of infected object                                                                 |                             |
```

Comment:
Do we need this as a param? Would it ever be desirable to quarantine under a different key to the original file? Could see this getting a little confusing

Reply:
Whilst you're are correct, the current behaviour does actually change the Key between the source and quarantine bucket:

- source key: `{user_id}/{consignment_id}/{file_id}`
- quarantine key: `{consignment_id}/{file_id}`

I suspect this was done because using the "user id" in the Key makes it harder to track down quarantined files. I'm very reluctant to refactor this too much as it's already complex, and it has implications beyond just this lambda